library-valcon-internship/src/components/BookForm/BookForm.tsx

```diff
@@ -0,0 +1,401 @@
+import BookFormValidation from '../../models/BookFormValidation'
+import AuthorForm from '../AuthorForm/AuthorForm'
+import './BookForm.css'
+
+interface BookFormProps {
+  bookForm: BookFormType
+  bookFormValidation: BookFormValidation
+  authorForm?: AuthorFormType
+  authorFormValidation?: AuthorFormValidation
+  setBookForm: Dispatch<SetStateAction<BookFormType>>
+  setBookFormValidation: Dispatch<SetStateAction<BookFormValidation>>
+  setAuthorForm?: Dispatch<SetStateAction<AuthorFormType>>
+  setAuthorFormValidation?: Dispatch<SetStateAction<AuthorFormValidation>>
+  handleOnAuthorSubmit?: () => void
+  handleOnBookSubmit: (e: SyntheticEvent) => void
+  title: string
+  handleShowAuthorModal?: () => void
+}
+
+const BookForm = ({
+  bookForm, bookFormValidation, authorForm, authorFormValidation,
+  setBookForm, setBookFormValidation, setAuthorForm, setAuthorFormValidation,
+  handleOnAuthorSubmit, handleOnBookSubmit, title, handleShowAuthorModal
+}: BookFormProps) => {
+  const hiddenFileInput = useRef<HTMLInputElement>(null)
+  const [ showAccordion, setShowAccordion ] = useState(false)
+
+  const handleCoverClick = () => {
+    if (hiddenFileInput.current)
+      hiddenFileInput.current.click()
+  }
+  const handleFileChange = ({ currentTarget }: FormEvent<HTMLInputElement>) => {
+    if (currentTarget.files) {
+      const files = currentTarget.files
+      const reader = new FileReader()
+      if (files) {
+        reader.readAsDataURL(files[0])
+        setBookForm(bookForm => {
+          return {
+            ...bookForm,
+            requestCover: files[0]
+          }
+        })
+        reader.onloadend = function () {
+          const base64data = reader.result
+          if (base64data)
+            setBookForm(bookForm => {
+              return {
+                ...bookForm,
+                cover: base64data as string
+              }
+            })
+        }
+      }
+    }
+  }
+
+  const handleOnChangeInput = (value: string, formProperty: string) => {
+    switch (formProperty) {
+      case 'Title':
+        setBookForm(bookForm => {
+          return {
+            ...bookForm,
+            title: value
+          }
+        })
+        break
+      case 'Description':
+        setBookForm(bookForm => {
+          return {
+            ...bookForm,
+            description: value
+          }
+        })
+        break
+      case 'Isbn':
+        setBookForm(bookForm => {
+          return {
+            ...bookForm,
+            isbn: value
+          }
+        })
+        break
+      case 'Release date':
+        setBookForm(bookForm => {
+          return {
+            ...bookForm,
+            releaseDate: new Date(value)
+          }
+        })
+        break
+      case 'Quantity':
+        setBookForm(bookForm => {
+          return {
+            ...bookForm,
+            quantity: value
+          }
+        })
+        break
+      default:
+        break
+    }
+  }
```

Comment:
again, no need for switch, you can access it with key: [formProperty.toLowerCase()]